### PR TITLE
Concat bugfix

### PR DESCRIPTION
**Pull request by DevNambi**
Fixing the concatenation bug identified by EmilyH. Converting the bash script awk/sed/cut code to Python+pandas, the result is a lot cleaner.

### CODE REVIEW

**Comment by sminot** [comment]
While we're making changes, it would be great to move the declaration of all default params into the nextflow.config file. This works out better in the long run, since we can modify those defaults without touching the code in `main.nf`

**Comment by sminot** [comment]
Looking over this code, I only have two comments, and they both happen to be related to the manipulation of CSV files.

One bit of code looked like it might be vulnerable to an odd edge case. Looking at `ConcatFiles_Round2`, I think that it might yield an incorrect output if it has input files with variable ordering of columns. Using a pattern like `pd.concat([pd.read_csv(fp) for fp in os.listdir('.')])` would account for that edge case.

There are a few other places where the CSV parsing on read/write seems overly cumbersome to me. For example:
```
    csv_dict = []
    with open(f"{src_dir}/{src_file}", 'r') as csvfile:
        reader = csv.DictReader(csvfile)
        for row in reader:
            csv_dict.append(row)
```
Could be replaced with `csv_dict = pd.read_csv(f"{src_dir}/{src_file}").to_dict(orient='records')`. I'm generally in favor of approaches which are more concise, and which use a common workhorse library like `pandas`.

File: main.nf [hunk 1]
```diff
@@ -14,14 +14,15 @@ params.group_col = 'Group_Number'
 params.folder_col = 'PathName_Orig'
 params.file_col = 'FileName_Orig'
 params.shard_col = "Shard_Id"
+params.experiment_file = 'Experiment.txt'
```
Comment (sminot) [inline main.nf]: With the default params included in `nextflow.config`, they can be removed here.

**Review by sminot — COMMENTED** [review]
The only substantive changes I would recommend would be to simplify the parsing and manipulation of CSV files, as described at more length https://github.com/FredHutch/cellprofiler-batch-nf/pull/5#issuecomment-1054767299

**Comment by DevNambi** [comment]
The upcoming feature request (for multichannel image support) will require refactoring the CSV-handling code in this workflow. I agree that using a workhorse library like pandas is optimal; I'm going to suggest we leave this version as it is, with the understanding that we'll be switching it to pandas with the upcoming feature.

I'll remove the default parameters from the `main.nf` script as you suggest.

**Comment by sminot** [comment]
Your plan makes a lot of sense to me. Better to wait to touch the CSV code if you're going to have to change it for the next feature anyway. Looks good to me!

**Pull request merged**